src/DGtal/images/ArrayImageAdapter.h

```diff
@@ -0,0 +1,544 @@
+   * This adapted image is writable iff the given iterator is mutable.
+   *
+   * The available iterators for this image can return the corresponding point and are
+   * faster than using an iterator over the domain (see ArrayImageIterator).
+   * Reverse iterators and ranges are defined in the inherited class IteratorCompletion.
+   *
+   * Some helpers are available (see makeArrayImageAdapterFromIterator and makeArrayImageAdapterFromImage)
+   * for easy construction (with template deduction) from an iterator or an CConstImage model.
+   *
+   * It is important to note that, since this class only adapts an already existing storage to an image, all
+   * copy operations (constructor and operator) lead to shallow copies.
+   *
+   * The following code snippet demonstrates how to use ArrayImageAdapter from converting a native C-array to an image:
+   * @snippet ../examples/images/exampleArrayImageAdapter.cpp ArrayImageAdapter_example
+   *
+   * Another usage example is illustrated in the image module documentation, \ref arrayimageadat.
+   *
+   * @remark The given random-access iterator can be either mutable or constant.
+   *
+   * @warning The array must be column-major ordered (but row-major order could be later accepted via template parameter, if needed ?)
+   * @warning The domain must be an HyperRectDomain.
+   * @warning C++11 needs to be enabled in order to use this class.
+   *
+   * @tparam TArrayIterator Type of a random-access iterator over the datas (can be a T* pointer).
+   * @tparam TSpace Type of the space associated to the HyperRectDomain (auto-deduced from TDomain template, see ArrayImageAdapter).
+   *
+   * @see makeArrayImageAdapterFromIterator
+   * @see makeArrayImageAdapterFromImage
+   * @see exampleArrayImageAdapter.cpp
+   */
+  template <
+    typename TArrayIterator,
+    typename TSpace
+  >
+  class ArrayImageAdapter< TArrayIterator, HyperRectDomain<TSpace> >
+      : public IteratorCompletion< ArrayImageAdapter< TArrayIterator, HyperRectDomain<TSpace> > >
+    {
+
+    // Checks Random-access iterator concept on TArrayIterator
+    BOOST_CONCEPT_ASSERT( (boost_concepts::RandomAccessTraversalConcept<TArrayIterator>) );
+
+    public:
+      // Typedefs
+      using Self = ArrayImageAdapter<TArrayIterator, HyperRectDomain<TSpace> >;           ///< Self type.
+      using ArrayIterator   = TArrayIterator;                                             ///< The given random-access iterator's type.
+      using Value           = typename std::iterator_traits<ArrayIterator>::value_type;   ///< The value type stored in the image.
+      using Reference       = typename std::iterator_traits<ArrayIterator>::reference;    ///< Mutable reference type.
+      using ConstReference  = const Reference;                                            ///< Constant reference type.
+
+      // DGtal typedefs and constant
+      using Domain    = HyperRectDomain<TSpace>;      ///< Domain type.
```

Comment:
My bad... I thought that doxygen append the type documentation ```//<``` in the public type section of the resulting HTML file. But it is not the case for ```typedefs``` as well